src/Microsoft.AspNet.Mvc.ModelBinding/Binders/ArrayModelBinder.cs

```diff
@@ -9,21 +9,19 @@ namespace Microsoft.AspNet.Mvc.ModelBinding
 {
     public class ArrayModelBinder<TElement> : CollectionModelBinder<TElement>
     {
-        public override Task<bool> BindModelAsync(ModelBindingContext bindingContext)
+        public override Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContext)
         {
             if (bindingContext.ModelMetadata.IsReadOnly)
             {
-                return Task.FromResult(false);
+                return Task.FromResult<ModelBindingResult>(null);
```

Comment:
Consider returning a `ModelBindingResult.Failed` result here instead of null. Slightly better abstraction.